collection-views-tests.js

```diff
@@ -258,6 +258,99 @@ Tinytest.add('where - upsert - t4 (no matches found, insert still works)', funct
   test.equal(typeof result.insertedId, 'string');
 });
 
+// Test if field specifiers work properly
+Tinytest.add('where - field specifiers - t1 (one specifier: only fetch the title)', function (test) {
+  var Books = newCollection();
+  var doc = Books
+    .where({ genre: 'science' }, { title: 1 })
+    .findOne({ kind: 'magazine' });
+  test.equal(doc.title, 'Science Magazine 1');
+  test.isNotUndefined(doc.title);
+  test.isUndefined(doc.genre);
+  test.isUndefined(doc.kind);
+  test.isUndefined(doc.catalogId);
+  test.isNotUndefined(doc._id);
+  test.isNotNull(doc._id);
+  test.equal(typeof doc._id, 'string');
+});
+
+Tinytest.add('where - field specifiers - t2 (two chained specifiers: only fetch the title and the kind)', function (test) {
+  var Books = newCollection();
+  var doc = Books
+    .where({}, { title: 1 })
+    .where({ genre: 'science' }, { kind: 1 })
+    .findOne({ kind: 'magazine' });
+  test.equal(doc.title, 'Science Magazine 1');
+  test.isNotUndefined(doc.title);
+  test.isUndefined(doc.genre);
+  test.isNotUndefined(doc.kind);
+  test.isUndefined(doc.catalogId);
+  test.isNotUndefined(doc._id);
+  test.isNotNull(doc._id);
+  test.equal(typeof doc._id, 'string');
+});
+
+Tinytest.add('where - field specifiers - t3 (one specifier: remove the title)', function (test) {
+  var Books = newCollection();
+  var doc = Books
+    .where({ genre: 'science' }, { title: 0 })
+    .findOne({ kind: 'magazine' });
+  test.isUndefined(doc.title);
+  test.isNotUndefined(doc.genre);
+  test.isNotUndefined(doc.kind);
+  test.isNotUndefined(doc.catalogId);
+  test.isNotUndefined(doc._id);
+  test.isNotNull(doc._id);
+  test.equal(typeof doc._id, 'string');
+});
+
+Tinytest.add('where - field specifiers - t4 (two specifier: remove the kind and catalogId)', function (test) {
```

Comment:
These tests could be more descriptively named. How about: `CollectionViews - field specifiers - chained where calls should restrict the set of fields`.

Also each test may contain more than one statement to be tested, for example this test could be written like so: 
```javascript:
var Books = newCollection();
var hideKindAndGenre = Books.where({}, {kind: 0}).where({}, {genre: 0});
test.equal(hideKindAndGenre.find().count(), Books.find().count());
test.isUndefined(hideKindAndGenre.findOne().kind);
test.isUndefined(hideKindAndGenre.findOne().genre);
var book = Books.findOne();
test.equal(hideKindAndGenre.findOne(book._id), _.omit(book, 'kind', 'genre'));
```